spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/batch/AbstractJdbcJobSearchableExecutionDaoTests.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.batch;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.batch.core.BatchStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobInstance;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.repository.dao.AbstractJdbcBatchMetadataDao;
+import org.springframework.batch.core.repository.dao.JdbcJobExecutionDao;
+import org.springframework.batch.core.repository.dao.JdbcStepExecutionDao;
+import org.springframework.batch.item.database.support.DataFieldMaxValueIncrementerFactory;
+import org.springframework.cloud.dataflow.core.database.support.DatabaseType;
+import org.springframework.cloud.dataflow.core.database.support.MultiSchemaIncrementerFactory;
+import org.springframework.cloud.task.batch.listener.support.JdbcTaskBatchDao;
+import org.springframework.cloud.task.repository.TaskRepository;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public abstract class AbstractJdbcJobSearchableExecutionDaoTests extends AbstractDaoTests {
+
+	private static final String BASE_JOB_INSTANCE_NAME = "JOB_INST_";
+
+	protected JdbcSearchableJobExecutionDao jdbcSearchableJobExecutionDao;
+
+	protected JdbcSearchableJobInstanceDao jdbcSearchableJobInstanceDao;
+
+	protected DataSource dataSource;
+
+	protected DataFieldMaxValueIncrementerFactory incrementerFactory;
+
+	protected JdbcJobExecutionDao jobExecutionDao;
+
+	protected JdbcStepExecutionDao stepExecutionDao;
+
+	protected JdbcTemplate jdbcTemplate;
+
+	protected TaskRepository taskRepository;
+
+	protected JdbcTaskBatchDao jdbcTaskBatchDao;
+
+	public void setupSearchableExecutionDaoTest(JdbcDatabaseContainer dbContainer, String schemaName,
+												DatabaseType databaseType) throws Exception {
+		this.dataSource = createDataSourceForContainer(dbContainer);
```

Comment:
**[This Comment Applies to all concrete impls]**

The pattern I am seeing is that a concrete subclass provides a testcontainer and then calls a base method via  `@BeforeEach` that initializes the DB to clean slate before running each test. 

The current impl requires each concrete test to do these things:
```java
this.dataSource = createDataSourceForContainer(dbContainer);
this.jdbcTemplate = new JdbcTemplate(this.dataSource);
createDataFlowSchema(dbContainer, schemaName);
```

I propose that we pull up the DS and JDBCT to the `AbstractDaoTests` and then provide a base method called `prepareForTest` that can be called by the concrete tests. Something like this: 
```java
protected void prepareForTest(JdbcDatabaseContainer dbContainer, String schemaName) {
	this.dataSource = createDataSourceForContainer(dbContainer);
	this.jdbcTemplate = new JdbcTemplate(this.dataSource);
	createDataFlowSchema(dbContainer, schemaName);
}
```
Otherwise, all the ADT is really providing is 2 util methods that everyone has to call.

Wdyt?